unsupported/Eigen/CXX11/src/Tensor/TensorReductionGpu.h

```diff
@@ -238,20 +240,6 @@ __global__ void FullReductionKernelHalfFloat(Reducer reducer, const Self input,
 
   // Initialize the output value if it wasn't initialized by the ReductionInitKernel
 
-#if defined(EIGEN_HIPCC)
-  
```

Comment:
@deven-amd could you explain the rationale why HIP-specific logic is not needed here anymore?

Reply:
Starting to look into this.

First trying to understand why the HIP specific change was needed in the first place.  Both the eigen unit tests and TF CI pass with the HIP specific code removed.